File: src/java/MyStack.java
package java;


// 用队列实现栈

import java.util.LinkedList;

class MyStack {

    private LinkedList<Integer> deque;

    /**
     * Initialize your data structure here.
     */
    public MyStack() {
        deque = new LinkedList<>();
    }

    /**
     * Push element x onto stack.
     */
    public void push(int x) {
        deque.offer(x);
        int size = deque.size();
        for (int i = 0; i < size - 1; i++) {

            deque.offer(deque.poll());


        }

    }

    /**
     * Removes the element on top of the stack and returns that element.
     */
    public int pop() {
        return deque.poll();
    }

    /**
     * Get the top element.
     */
    public int top() {
        return deque.peek();
    }

    /**
     * Returns whether the stack is empty.
     */
    public boolean empty() {
        return deque.isEmpty();
    }
}

File: src/design/MyStack.kt
package design

class MyStack() {

    /** Initialize your data structure here. */

    var deque = ArrayDeque<Int>()

    /** Push element x onto stack. */
    fun push(x: Int) {
        deque.addLast(x)
        var size = deque.size
        for (i in 1 until size) {
            deque.addLast(deque.removeFirst())
        }


    }

    /** Removes the element on top of the stack and returns that element. */
    fun pop(): Int {
        return deque.removeFirst()
    }

    /** Get the top element. */
    fun top(): Int {
        return deque.first()
    }

    /** Returns whether the stack is empty. */
    fun empty(): Boolean {
        return deque.isEmpty()
    }

}